include/kernel_structs.h

```diff
@@ -121,7 +122,7 @@ struct _cpu {
 
 #ifdef CONFIG_TIMESLICING
 	/* number of ticks remaining in current time slice */
-	int slice_ticks;
+	k_ticks_t slice_ticks;
```

Comment:
Just mentioned this one in a reply to my own PR, but this we want to be int32_t.  A timeslice is specified at the API level in tick-appropriate units (32 bit ms, or a count of ticks for the newer per-thread API).  We don't want to pay the code size overhead for 64 bit ticks here.

Reply:
That is reasonable, but is it a considerable overhead ? We are talking about additional bytes in a struct per core. Of course, there are other changes in some functions to use `k_ticks_t` that can increase the stack usage, but that shouldn't be that much ...